common/include/common/framebuffer.h

```diff
@@ -24,9 +24,23 @@
 #include <stdlib.h>
 #include <stdbool.h>
 #include <stdint.h>
+#include <stdatomic.h>
 
 typedef struct stFrameBuffer FrameBuffer;
 
+#define FB_CHUNK_SIZE   1048576 // 1MB
+#define FB_SPIN_LIMIT   10000   // 10ms
+#define FB_WP_TYPE      atomic_uint_least32_t
+#define FB_WP_SIZE      sizeof(FB_WP_TYPE)
+
+struct stFrameBuffer
+{
+  FB_WP_TYPE wp;
+  uint8_t    data[0];
+};
+
+static const size_t FrameBufferStructSize = sizeof(FrameBuffer);
```

Comment:
At this point we might as well get rid of this and just inline `sizeof(FrameBuffer)`, since `FrameBufferStructSize` only existed as a hack to avoid exposing `struct stFrameBuffer`.